build.sbt

```diff
@@ -178,6 +179,11 @@ lazy val server = (project in file("server"))
     javaOnlyReleaseSettings,
     javafmtCheckSettings,
     javaCheckstyleSettings(file("dev") / "checkstyle-config.xml"),
+    Compile / compile / javacOptions ++= Seq(
+      "--release", "17",
+      "-processor",
+      "lombok.launch.AnnotationProcessorHider$AnnotationProcessor"
```

Comment:
what is this line for? can write that down as a comment?

Reply:
I think @ravivj-db had added this but from ChatGPT: This is essentially instructing the Java compiler to use Lombok's annotation processor (AnnotationProcessorHider$AnnotationProcessor) to process the annotations present in the code. This allows Lombok to generate the necessary boilerplate code (like getters, setters, etc.) during the compilation phase.

Typically, we don’t need to specify this option manually as it’s automatically handled by build tools like Maven or Gradle when you include Lombok as a dependency.

However, maybe sbt doesn't handle that hence this was needed.